audience-demo/src/App.js

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { useState, useCallback } from "react";
+import { AudienceDisplay } from "./AudienceDisplay";
+import { SelectUser } from "./SelectUser"
+
+export const App = () => {
+  const [displayAudience, setDisplayAudience] = useState(false);
+  const [userId, setUserId] = useState();
+  const [containerId, setContainerId] = useState();
+
+  const handleSelectUser = useCallback((userId, containerId) => {
+    setDisplayAudience(true)
+    setUserId(userId);
+    setContainerId(containerId);
+  });
+
+  const handleContainerNotFound = useCallback(() => {
+    setDisplayAudience(false)
+  });
```

Comment:
when you're using useCallback I believe you're supposed to connect state changes that should re-trigger 

```suggestion
  const handleSelectUser = useCallback((userId, containerId) => {
    setDisplayAudience(true)
    setUserId(userId);
    setContainerId(containerId);
  }, [displayAudience, userId, containerId]);

  const handleContainerNotFound = useCallback(() => {
    setDisplayAudience(false)
  }, [displayAudience]);
```